src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -1110,6 +1110,29 @@ private StringBuilder getRowsOfPossibleRecipientsInCsvFormat(
         }
         return sessionsToSendEmailsFor;
     }
+    
+    /**
+     * Criteria: must be published, publishEmail must be enabled and
+     * resultsVisibleTime must be custom.
+     * 
+     * @return returns a list of sessions that require automated emails to be
+     *         sent as they are published
+     */
+    public List<FeedbackSessionAttributes> getFeedbackSessionsWhichNeedAutomatedUnpublishedEmailsToBeSent() {
+        List<FeedbackSessionAttributes> sessions =
+                fsDb.getFeedbackSessionsWithUnsentUnpublishedEmail();
+        List<FeedbackSessionAttributes> sessionsToSendEmailsFor =
+                new ArrayList<FeedbackSessionAttributes>();
+
+        for (FeedbackSessionAttributes session : sessions) {
+            // automated emails are required only for custom unpublish times
+            if (!session.isPublished() && session.isPublishedEmailEnabled()
```

Comment:
yeah, we can do this with `sentPublishedEmail` only. Making an extra field was not at all necessary. Using  `sentPublishedEmail`, will also ensure the sending of unpublished emails only if published emails has been sent.

But I don't get the the idea of adding code to `InstructorFeedbackUnpublishAction`. It will only handle custom unpublished sessions ?